src/renderer/views/Channel/Channel.js

```diff
@@ -453,13 +460,19 @@ export default defineComponent({
 
     // Enable auto refresh on sort value change AFTER initial update done
     if (!process.env.SUPPORTS_LOCAL_API || this.backendPreference === 'invidious') {
-      this.getChannelInfoInvidious()
-      this.autoRefreshOnSortByChangeEnabled = true
+      await this.getChannelInfoInvidious().finally(() => {
+        this.autoRefreshOnSortByChangeEnabled = true
+      })
     } else {
-      this.getChannelLocal().finally(() => {
+      await this.getChannelLocal().finally(() => {
         this.autoRefreshOnSortByChangeEnabled = true
       })
     }
+
+    const oldQuery = this.$route.query.searchQueryText ?? ''
+    if (oldQuery !== null && oldQuery !== '') {
+      this.newSearch(oldQuery)
+    }
```

Comment:
If you figure out what's messing with the query result sort order let me know
I read the rest of the code I don't think FT even tries to mess with the result sort order